greet.py

```diff
@@ -1,19 +1,54 @@
 def greet(*names):
-    if len(names) == 0:
+    names_len = len(names)
+    if names_len == 0:
         string = 'Hello, my friend'
         return string
+    shouted_names = []
+    simple_names = []
+    for name in names:
+        if not isinstance(name, str):
+            raise ValueError("Wrong input")
+        if name.isupper():
+            shouted_names.append(name)
+        else:
+            simple_names.append(name)
+    if names_len == 1:
+        if names[0].isupper():
+            string = 'HELLO, ' + names[0]
+        else:
+            string = 'Hello, ' + names[0]
     else:
-        for name in names:
-            if not isinstance(name, str):
-                string = 'Wrong input'
-                return string
-        if len(names) == 1:
-            if names[0].isupper():
-                string = 'HELLO, ' + names[0]
-            else:
-                string = 'Hello, ' + names[0]
+        j = 0
+        simple_names_len = len(simple_names)
+        shouted_names_len = len(shouted_names)
+        string = 'Hello '
+        if simple_names_len == 1:
+            string += ', ' + simple_names[0] + '.'
+            string = string.replace(' ', '', 1)
         else:
-            string = 'Hello, ' + names[0] + ' and ' + names[1]
-    return string
+            for simple_name in simple_names:
+                string += simple_name + ' '
+            string = string.strip()
+            string_list = list(reversed(string.split()))
+            string = ' and '.join(string_list)
+            string = ' '.join(list(reversed(string.split())))
+            string = string.replace(' and ', ', ', simple_names_len - 1)
+            string += '.'
 
+        if shouted_names_len != 0:
```

Comment:
what's happening for shouted names look very similar as for simple ones. Is that possible to extract the logic to a function?